packages/@tailwindcss-vite/src/index.ts

```diff
@@ -16,23 +18,19 @@ export default function tailwindcss(): Plugin[] {
   let scanner: Scanner | null = null
   let changedContent: { content: string; extension: string }[] = []
   let candidates = new Set<string>()
-  let fullRebuildPaths: string[] = []
+  let fullRebuildPaths = new Set<string>()
+  let rebuildStrategy: 'full' | 'incremental' = 'full'
+  let compiler: null | Awaited<ReturnType<typeof compile>> = null
```

Comment:
The compiler is now persisted between rebuilds. We pipe every change through a `onFileChange` function that will compare it with the list of `fullRebuildPaths`. If a change is detected, it will set the `rebuildStrategy` for the next emitted build to `full`.

Todo for myself: The compiler and the `rebuildStrategy` should probably be keyed by the input css file ID in case there are multiple in one Vite project.

Reply:
I thought about it more and it should _definitely_ be keyed by the CSS input ID. In fact I think this should be called `root` or so and should also replace the current `cssModules` var. 

I need to find out exactly why we have the `handled` flag though.